ai2thor/tests/test_unity.py

```diff
@@ -843,6 +836,27 @@ def test_get_scenes_in_build(controller):
     assert len(diff) == 0, "scenes in build diff: %s" % diff
 
 
+@pytest.mark.parametrize("controller", [wsgi_controller, fifo_controller])
+def test_get_reachable_positions(controller):
+    event = controller.step("GetReachablePositions")
+    assert (
+        event.metadata["actionReturn"] == event.metadata["reachablePositions"]
+    ), "reachablePositions should map to actionReturn!"
+    assert len(event.metadata["reachablePositions"]) > 0 and isinstance(
+        event.metadata["reachablePositions"], list
+    ), "reachablePositions/actionReturn should not be empty after calling GetReachablePositions!"
+
+    assert "reachablePositions" not in event.metadata.keys()
+    event = controller.step("Pass")
+    try:
+        event.metadata["reachablePositions"]
+        assert (
+            False
+        ), "reachablePositions shouldn't be available without calling action='GetReachablePositions'."
+    except:
+        pass
```

Comment:
There are 3 tests happening here, each of which is intentional:
1. Upon calling `GetReachablePositions`, calling `event.metadata["reachablePositions"]` should silently alias to `actionReturn` (which contains the reachable positions).
2. `reachablePositions` should never appear as a key in `event.metadata` because it is now deprecated (in favor of using `actionReturn` directly). However, calling `event.metadata["reachablePositions"]` will now be sideways and backwards compatible.
3. If a user does not call `GetReachablePositions` and tries to access `event.metadata["reachablePositions"]`, instead of the old way that returned an empty list, an `IndexError` is thrown. This now removes that silent bug.